geth/transactions/txqueue_manager_test.go

```diff
@@ -83,41 +119,72 @@ func (s *TxQueueTestSuite) TestCompleteTransaction() {
 		Address:    common.FromAddress(TestConfig.Account1.Address),
 		AccountKey: &keystore.Key{PrivateKey: key},
 	}
-	s.setupStatusBackend(account, password, nil)
 
 	nonce := hexutil.Uint64(10)
 	gas := hexutil.Big(*big.NewInt(defaultGas + 1))
-	s.setupTransactionPoolAPI(account, nonce, gas, nil)
 
-	txQueueManager := NewManager(s.nodeManagerMock, s.accountManagerMock)
-
-	txQueueManager.Start()
-	defer txQueueManager.Stop()
-
-	tx := common.CreateTransaction(context.Background(), common.SendTxArgs{
-		From: common.FromAddress(TestConfig.Account1.Address),
-		To:   common.ToAddress(TestConfig.Account2.Address),
-	})
-
-	s.NoError(txQueueManager.QueueTransaction(tx))
-	w := make(chan struct{})
-	var (
-		hash gethcommon.Hash
-		err  error
-	)
-	go func() {
-		hash, err = txQueueManager.CompleteTransaction(tx.ID, password)
-		s.NoError(err)
-		close(w)
-	}()
+	testCases := []struct {
+		name     string
+		gas      *hexutil.Big
+		gasPrice *hexutil.Big
+	}{
+		{
+			"noGasDef",
+			nil,
+			nil,
+		},
+		{
+			"gasDefined",
+			(*hexutil.Big)(big.NewInt(100)),
+			nil,
+		},
+		{
+			"gasPriceDefined",
+			nil,
+			(*hexutil.Big)(big.NewInt(100)),
+		},
+	}
 
-	rst := txQueueManager.WaitForTransaction(tx)
-	// Check that error is assigned to the transaction.
-	s.NoError(rst.Error)
-	// Transaction should be already removed from the queue.
-	s.False(txQueueManager.TransactionQueue().Has(tx.ID))
-	s.NoError(WaitClosed(w, time.Second))
-	s.Equal(hash, rst.Hash)
+	for _, testCase := range testCases {
+		s.T().Run(testCase.name, func(t *testing.T) {
```

Comment:
never used it with testify, will it allow to use testify.m the same way as before?

Reply:
Looks like it doesn't detect that. But IMHO, it's certainly more readable & DRY and all tests run in few seconds anyway.

Reply:
looks good i think

> go test ./geth/transactions/ -v -testify.m=TestCompleteTransaction$
=== RUN   TestTxQueueTestSuite
=== RUN   TestTxQueueTestSuite/TestCompleteTransaction
=== RUN   TestTxQueueTestSuite/TestCompleteTransaction/noGasDef
=== RUN   TestTxQueueTestSuite/TestCompleteTransaction/gasDefined
=== RUN   TestTxQueueTestSuite/TestCompleteTransaction/gasPriceDefined
--- PASS: TestTxQueueTestSuite (0.01s)
    --- PASS: TestTxQueueTestSuite/TestCompleteTransaction (0.01s)
        --- PASS: TestTxQueueTestSuite/TestCompleteTransaction/noGasDef (0.00s)
        --- PASS: TestTxQueueTestSuite/TestCompleteTransaction/gasDefined (0.00s)
        --- PASS: TestTxQueueTestSuite/TestCompleteTransaction/gasPriceDefined (0.00s)
PASS
ok  	github.com/status-im/status-go/geth/transactions	0.055s